libs/GLMLoader/GLMLoader.cpp

```diff
@@ -0,0 +1,710 @@
+	//m_indices = (GLuint *)malloc(sizeof(GLuint) * 3 * numtriangles);
+	//m_numvertices = 3 * numtriangles;
+
+	std::vector<glm::vec3> v_vertices(sizeof(glm::vec3) * numvertices);
+	std::vector<glm::vec3> v_normals(sizeof(glm::vec3) * numnormals);
+	std::vector<glm::vec2> v_textures(sizeof(glm::vec2) * numsUV);
+	
+
+	std::vector<unsigned int> v_indices(3 * numtriangles);
+
+	for (unsigned int i = 0; i < 3 * numtriangles; ++i)
+	{
+		v_indices[i] = i;
+	}
+
+	    int vertexCount = 0;
+		int normalCount = 0;
+		int textureCount = 0;
+		for (unsigned int i = 0; i < numtriangles; i++)
+		{
+			glm::vec3 vertex1;
+			vertex1.x = vertices[3 * (triangles[i].vindices[0] - 1) + 0];
+			vertex1.y = vertices[3 * (triangles[i].vindices[0] - 1) + 1];
+			vertex1.z = vertices[3 * (triangles[i].vindices[0] - 1) + 2];
+			v_vertices[vertexCount++] = vertex1;
+
+			glm::vec3 vertex2;
+			vertex2.x = vertices[3 * (triangles[i].vindices[1] - 1) + 0];
+			vertex2.y = vertices[3 * (triangles[i].vindices[1] - 1) + 1];
+			vertex2.z = vertices[3 * (triangles[i].vindices[1] - 1) + 2];
+			v_vertices[vertexCount++] = vertex2;
+
+			glm::vec3 vertex3;
+			vertex3.x = vertices[3 * (triangles[i].vindices[2] - 1) + 0];
+			vertex3.y = vertices[3 * (triangles[i].vindices[2] - 1) + 1];
+			vertex3.z = vertices[3 * (triangles[i].vindices[2] - 1) + 2];
+			v_vertices[vertexCount++] = vertex3;
+
+			if (numnormals > 0)
+			{
+				glm::vec3 normal1;
+				normal1.x = normals[3 * (triangles[i].nindices[0] - 1) + 0];
+				normal1.y = normals[3 * (triangles[i].nindices[0] - 1) + 1];
+				normal1.z = normals[3 * (triangles[i].nindices[0] - 1) + 2];
+				v_normals[normalCount++] = normal1;
+
+				glm::vec3 normal2;
+				normal2.x = normals[3 * (triangles[i].nindices[1] - 1) + 0];
+				normal2.y = normals[3 * (triangles[i].nindices[1] - 1) + 1];
+				normal2.z = normals[3 * (triangles[i].nindices[1] - 1) + 2];
+				v_normals[normalCount++] = normal2;
+
+				glm::vec3 normal3;
+				normal3.x = normals[3 * (triangles[i].nindices[2] - 1) + 0];
+				normal3.y = normals[3 * (triangles[i].nindices[2] - 1) + 1];
+				normal3.z = normals[3 * (triangles[i].nindices[2] - 1) + 2];
+				v_normals[normalCount++] = normal3;
+			}
+			
+			if (numsUV > 0)
+			{
+				glm::vec2 uv1;
+				uv1.x = texcoords[2 * (triangles[i].txindices[0] - 1) + 0];
+				uv1.y = texcoords[2 * (triangles[i].txindices[0] - 1) + 1];
+				v_textures[textureCount++] = uv1;
+				
+				glm::vec2 uv2;
+				uv2.x = texcoords[2 * (triangles[i].txindices[1] - 1) + 0];
+				uv2.y = texcoords[2 * (triangles[i].txindices[1] - 1) + 1];
+				v_textures[textureCount++] = uv2;
+
+				glm::vec2 uv3;
+				uv3.x = texcoords[2 * (triangles[i].txindices[2] - 1) + 0];
+				uv3.y = texcoords[2 * (triangles[i].txindices[2] - 1) + 1];
+				v_textures[textureCount++] = uv3;
+			}
+
+
+		}
+	
+
+
+
+	free(vertices);
+	free(normals);
+	free(triangles);
+	free(texcoords);
+
+
+
+
+	VertexBuffer* vertexObject = new VertexBuffer();
```

Comment:
this is a really big function, and parts of the logic seems (to my naive eye) a bit repetitive - could some be pulled out into utility functions?

Reply:
This function  is not been used whatsoever. I removed it.